src/diffusers/loaders/single_file.py

```diff
@@ -247,71 +230,142 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
         local_files_only = kwargs.pop("local_files_only", False)
         revision = kwargs.pop("revision", None)
         torch_dtype = kwargs.pop("torch_dtype", None)
+        load_connected_pipeline = kwargs.pop("load_connected_pipeline", False)
+        local_dir = kwargs.pop("local_dir", None)
+        local_dir_use_symlinks = kwargs.pop("local_dir_use_symlinks", "auto")
 
-        class_name = cls.__name__
-
-        original_config, checkpoint = fetch_ldm_config_and_checkpoint(
-            pretrained_model_link_or_path=pretrained_model_link_or_path,
-            class_name=class_name,
-            original_config_file=original_config_file,
+        checkpoint = load_single_file_model_checkpoint(
+            pretrained_model_link_or_path,
             resume_download=resume_download,
             force_download=force_download,
             proxies=proxies,
             token=token,
-            revision=revision,
-            local_files_only=local_files_only,
             cache_dir=cache_dir,
+            local_files_only=local_files_only,
+            revision=revision,
+            local_dir=local_dir,
+            local_dir_use_symlinks=local_dir_use_symlinks,
         )
+        if original_config_file is not None:
+            original_config = fetch_original_config(
+                checkpoint, original_config_file, local_files_only=local_files_only
+            )
+        else:
+            original_config = None
 
         from ..pipelines.pipeline_utils import _get_pipeline_class
 
         pipeline_class = _get_pipeline_class(
             cls,
             config=None,
             cache_dir=cache_dir,
+            load_connected_pipeline=load_connected_pipeline,
         )
+        default_pipeline_config = fetch_diffusers_config(checkpoint)
+        config_file = hf_hub_download(
+            default_pipeline_config["pretrained_model_name_or_path"],
+            filename=cls.config_name,
+            cache_dir=cache_dir,
+            revision=revision,
+            proxies=proxies,
+            force_download=force_download,
+            resume_download=resume_download,
+            token=token,
+            local_files_only=local_files_only,
+        )
```

Comment:
There are many single file checkpoints for which we don't have equivalent diffusers checkpoints, but they can be loaded into a pipeline. How does this stand in that case? We fetch the corresponding config file examining the checkpoint and do a best effort to map accordingly? If so, I think there should be a comment about it.

Reply:
do you have examples of these? 
>There are many single file checkpoints for which we don't have equivalent diffusers checkpoints

Reply:
Have seen many but cannot recall any exact names of the top of my head. You can find a couple on CivitAI too (it hosts full finetunes of SD and SDXL too not just LoRAs). Cc: @vladmandic here.